src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -452,5 +498,35 @@ private async Task<List<string>> ExecuteCommandAsync(IExecutionContext context,
             return outputs;
         }
 #endif
+
+        private async Task CreateContainerNetworkAsync(IExecutionContext executionContext, string network)
+        {
+            Trace.Entering();
+            ArgUtil.NotNull(executionContext, nameof(executionContext));
+            int networkExitCode = await _dockerManger.DockerNetworkCreate(executionContext, network);
+            if (networkExitCode != 0)
+            {
+                throw new InvalidOperationException($"Docker network create failed with exit code {networkExitCode}");
+            }
+            // Expose docker network to env
+            executionContext.Variables.Set(Constants.Variables.Agent.ContainerNetwork, network);
+        }
+
+        private async Task RemoveContainerNetworkAsync(IExecutionContext executionContext, string network)
+        {
+            Trace.Entering();
+            ArgUtil.NotNull(executionContext, nameof(executionContext));
+            ArgUtil.NotNull(network, nameof(network));
+
+            executionContext.Output($"Remove container network: {network}");
+
+            int removeExitCode = await _dockerManger.DockerNetworkRemove(executionContext, network);
+            if (removeExitCode != 0)
+            {
+                executionContext.Warning($"Docker network rm failed with exit code {removeExitCode}");
+            }
+            // Remove docker network from env
+            executionContext.Variables.Set(Constants.Variables.Agent.ContainerNetwork, null);
```

Comment:
do we have to unset variable?

Reply:
Probably not, but it is technically no longer set after the container is removed. It is very unlikely that it will be accessed between removing container and finishing the job though